File: ToolboxComparison/saveraw.m
%load directory in question (change this to match your file structure)
imgdir = dir('./*/*labelsynced.nii');

%go through directory
for i = 1 : length(imgdir)
    
    %this matches ID in my file names
    file_name = imgdir(i).name(1:end-4);

    %load nii
    imgs = load_untouch_nii(fullfile(imgdir(i).folder, imgdir(i).name)).img;
    
    %get size
    [xx,yy,zz] = size(imgs);
    
    %save as raw type double
    fid=fopen(strcat(file_name, '.raw'),'w+');
    fwrite(fid,imgs,'double');
    fclose(fid);


end

%Test that it worked by opening last raw file:
fid = fopen(strcat(imgdir(i).name(1:end-4),'.raw'));
rawdata = fread(fid, inf, '*double');
rawdata = reshape(rawdata, xx, yy, zz);

figure; imshow(rawdata(:,:,floor(zz/2)), []);

File: Nii_Raw_Interconversion/saveraw.py
import os
import nibabel as nib
import numpy as np
import matplotlib.pyplot as plt

# set directory path
directory = "./"

# get all files ending in '*labelsynced.nii' in the directory and its subdirectories
imgdir = [os.path.join(root, name) for root, dirs, files in os.walk(directory) for name in files if name.endswith('*labelsynced.nii')]

# go through directory
for i in range(len(imgdir)):
    # this matches ID in my file names
    file_name = os.path.splitext(os.path.basename(imgdir[i]))[0]

    # load nii
    imgs = nib.load(imgdir[i]).get_fdata()

    # get size
    xx, yy, zz = imgs.shape

    # save as raw type double
    with open(file_name + '.raw', 'wb') as f:
        f.write(imgs.astype(np.float64).tobytes())

    # test that it worked by opening last raw file
    with open(file_name + '.raw', 'rb') as f:
        rawdata = np.frombuffer(f.read(), dtype=np.float64)
        rawdata = rawdata.reshape(xx, yy, zz)

    plt.imshow(rawdata[:, :, zz//2], cmap='gray')
    plt.show()
